### ISSUE TITLE
1.1.0: "ClientFunction cannot return DOM elements" when using client function to get a property of window

### ISSUE THREAD

**Issue by jcollum-nike** [DevExpress/testcafe#3633]
### What is your Test Scenario?
I want to check the existence of a window object
 
### What is the Current behavior?
Error: ClientFunction cannot return DOM elements. Use Selector functions for this purpose

### What is the Expected behavior?
The object should be returned, since it is a valid object. 

Note that replacing `getNavDispatch` with `getWindowLocation` in the test does not result in the error. They are both valid properties of `window`

If I add `await t.debug()` at the top of the test then open a console:

![image](https://user-images.githubusercontent.com/40277227/55445957-89520f80-5572-11e9-88b7-371106552449.png)

 
### What is your web application and your TestCafe test code?

 
nike.com/orders

<details>
<summary>Your complete test code (or attach your test files):</summary>

 
```js

const getNavDispatch = ClientFunction(() => window.NikeDotcomNav);
const getWindowLocation = ClientFunction(() => window.location);

fixture`Login`
  .httpAuth({
    username: process.env.BASIC_AUTH_USER,
    password: process.env.BASIC_AUTH_PASS,
  })
  .page(`${page.settings.baseUrl}/orders/`)
  .beforeEach(async () => {
    await waitForReact();
  });


test('user can log in', async t => {
  // Login via guest lookup page.
  const disp = await getNavDispatch();
  console.log('NikeDotcomNav', disp);
...
 
```
</details>

<details>
<summary>Your complete test report:</summary>

```
Login
 ✖ user can log in

   1) ClientFunction cannot return DOM elements. Use Selector functions for this purpose.

      Browser: Chrome 73.0.3683 / Mac OS X 10.13.6

```
</details>
 
### Your Environment details:
 
* testcafe version:                  1.1.0
* node.js version:                    8.15.1
* command-line arguments:  `NODE_ENV=development testcafe -s screenshots --skip-js-errors chrome ./tests/*.test.js -F Login`
* browser name and version: 
* platform and version:          OSX

**Comment by jcollum-nike** [comment]
If anyone is curious the fix for this was to JSON.stringify the object I wanted to get back. Ran into a few problems with that (circular structure). The root of the problem is related to object marshalling across processes -- I didn't need the full object, I just needed to check for its existence. 

Still, the error message I got has nothing to do with that. It could be improved.

**Comment by AndreyBelym** [comment]
I've found that the `NikeDotcomNav` contains at least one DOM element. Check the `window.NikeDotcomNav.vanilla.objects[1].el` property.

![image](https://user-images.githubusercontent.com/4479386/55565742-0f9d5b80-5703-11e9-8744-8434c93e2943.png)

However, I understand that the error message is very unclear and DOM elements can be missed easily when you try to retrieve an object with a very complex structure. I think we should display the name of the property that causes the error and names of its parent properties as well. I've created a feature suggestion about it: https://github.com/DevExpress/testcafe/issues/3640.
